### DISCUSSION TITLE
Is it ungoogled? Telemetry, home calls exist?

### DISCUSSION THREAD

**Discussion by Flashwalker** [Alex313031/thorium#119] [Q&A]
Is it debloated from google telemetry, home calls, etc?

**Comment by gz83** [comment]
> Is it debloated from google telemetry, home calls, etc?

@Flashwalker 

similar problem:
https://github.com/Alex313031/thorium/issues/115

**Comment by Alex313031** [comment]
@Flashwalker It has patches from ungoogles chromium, as well as patches that remove google telemetry and "phoning home" as much as possible, but I did not want to sacrifice usability, so google search, google sync, and the webstore are still enabled, and when/if you use those services, then thorium will talk to google servers in the same fashion as any chromium based browser that has these integrations.

**Comment by Flashwalker** [comment]
Well... At very first launch it continuously trying to connect to Google (despite none url was opened)

![thorium-very-first-start](https://user-images.githubusercontent.com/2188215/224488285-e84e5032-dc28-4779-92fa-4c68606880b0.jpg)

so it's not ungoogled enough.

**Reply by gz83** [reply]
> Well... At very first launch it continuously trying to connect to Google (despite none url was opened)
> 
> ![thorium-very-first-start](https://user-images.githubusercontent.com/2188215/224488285-e84e5032-dc28-4779-92fa-4c68606880b0.jpg)
> 
> so it's not ungoogled enough.

@Flashwalker 

Addition to Alex's answer: Thorium still retains some of Google's services, so it is normal to connect to Google's servers.  In addition, we just selected some patches from ungoogled-chromium, and you can find the specific patch list in the readme.

**Comment by Alex313031** [comment]
@Flashwalker @gz83 Those connections you are seeing are to preload the NTP, and to check if the user has enabled sync.

**Comment by Flashwalker** [comment]
After turning off everything that can be turned off in settings about account, sync, dns, "security", etcetera
it's still google-addict.
![2023-03-12_07-47-27](https://user-images.githubusercontent.com/2188215/224518320-a86c7ec6-1228-4f76-8fac-258f8ed1afd7.jpg)
So it doesn't fit my paranoia.

**Reply by Username23453** [reply]
@Alex313031, @Flashwalker , @gz83 ,whats it trying to connect to in this last screenshot?
(Thanks for you comments @Flashwalker )

**Reply by gz83** [reply]
> @Alex313031, @Flashwalker , @gz83 ,whats it trying to connect to in this last screenshot? (Thanks for you comments @Flashwalker )

Google's update service, account service and optimization guide

**Reply by Username23453** [reply]
@gz83 , can we have toggle buttons to turn all these off? To prevent tracking,etc?  I think thats important. And what do u mean by "optimization guide", why is google contacting that optimization guide (server?)?

**Reply by gz83** [reply]
> @gz83 , can we have toggle buttons to turn all these off? To prevent tracking,etc? I think thats important. And what do u mean by "optimization guide", why is google contacting that optimization guide (server?)?

@Intellectual234 

Explanation from chromium:

`The optimization guide component contains code for processing hints and machine learning models received from the remote Chrome Optimization Guide Service.

Optimization Guide is a layered component (https://sites.google.com/a/chromium.org/dev/developers/design-documents/layered-components-design) to enable it to be easily used on all platforms.

Directory structure: core/: Shared code that does not depend on src/content/

Contains the core functionalities to fetch and persist data received from the remote Chrome Optimization Guide Service, including but not limited to page load metadata and machine learning models. content/: Driver for the shared code based on the content layer
Contains the functionality for interpreting the data received from the remote Chrome Optimization Guide Service.`

**Comment by Jarmund5** [comment]
Why does it seriously need to keep this project "googled" ? sorry to revive an old thread but why can't thorium just be "a faster ungoogled chromium fork" ? (i mean this by all the patches and compiler optimizations) @Alex313031 i wish to know your reasoning behind keeping your project in this state.

**Reply by liumingedwin** [reply]
You can apply these patches on your own:(    
Thorium is simply a Chromium fork with optimizations.
